tornado_rest_jsonapi/resource.py

```diff
@@ -173,42 +164,35 @@ def post(self):
         if errors:
             raise exceptions.BadRequest(errors_from_jsonapi_errors(errors))
 
-        identifier = yield data_layer.create_object(data)
-
-        self._send_created_to_client(identifier)
-
-    @gen.coroutine
-    def put(self):
-        """You cannot PUT on a collection"""
-        raise HTTPError(int(http.client.METHOD_NOT_ALLOWED))
+        obj = yield data_layer.create_object(data, view_kwargs)
+        result = schema.dump(obj).data
 
-    @gen.coroutine
-    def delete(self):
-        raise HTTPError(int(http.client.METHOD_NOT_ALLOWED))
+        location = result['data']['links']['self']
+        self._send_created_to_client(location)
 
 
 class ResourceDetails(Resource):
     """Handler for URLs addressing a resource.
     """
     @gen.coroutine
-    def get(self, identifier):
+    def get(self, *args, **view_kwargs):
         """Retrieves the resource representation."""
-        data_layer = self.get_data_layer()
+        data_layer = self.get_data_layer_instance()
         qs = QSManager(self.request.arguments, self.schema)
         schema = compute_schema(self.schema,
                                 {},
                                 qs,
                                 qs.include)
 
-        obj = yield data_layer.retrieve_object(identifier)
+        obj = yield data_layer.get_object(view_kwargs)
```

Comment:
Tested with devour, it doesn't work. The identifier is passed as an arg, not as a kwarg. 

Javascript side:
```javascript
devourJsonApi.find('application', 5)
```
Python side:
```python
obj = yield data_layer.get_object(args[0])  # arg[0] === 5
```
My ideas are not clear enough to know where the issue could come from... But we will have the same issue for each method (get/patch/delete)